bin/sensei.js

```diff
@@ -167,4 +160,33 @@ process.on("unhandledRejection", (err) => {
   throw err;
 });
 
-cli(process.argv);
+cli(
+  yargs(process.argv.slice(2))
+    .command("serve")
+    .command("pdf")
+    .option("material", {
```

Comment:
`material` and `slug` are positional arguments, but are now documented as options in the usage, which is wrong:

```
sensei.js <command>

Commands:
  sensei.js serve [material] [slug]
  sensei.js pdf

Options:
  --version      Show version number                                   [boolean]
  --material     Path to the folder containing the training material.
                                 [string] [default: "current working directory"]
  --slug         Training name used in PDF files and HTML page titles.
                         [string] [default: "name of current working directory"]
  --slideWidth   Forwarded to Reveal.js. See
                 https://revealjs.com/presentation-size/.
                                                        [number] [default: 1420]
  --slideHeight  Forwarded to Reveal.js. See
                 https://revealjs.com/presentation-size/.[number] [default: 800]
  --help         Show help                                             [boolean]
```

Reply:
And are even rejected:

```shell-session
$ sensei_local pdf . test-training
sensei.js pdf

Options:
  --version      Show version number                                   [boolean]
  --material     Path to the folder containing the training material.
                                 [string] [default: "current working directory"]
  --slug         Training name used in PDF files and HTML page titles.
                         [string] [default: "name of current working directory"]
  --slideWidth   Forwarded to Reveal.js. See
                 https://revealjs.com/presentation-size/.
                                                        [number] [default: 1420]
  --slideHeight  Forwarded to Reveal.js. See
                 https://revealjs.com/presentation-size/.[number] [default: 800]
  --help         Show help                                             [boolean]

Unknown arguments: ., test-training
```

Reply:
This PR changes argument passing. Both material and slug are not positional arguments any more.

Reply:
But changing the CLI will break tools that are already using with the form as positional arguments.

Maybe it's out of scope of this PR to do more than adding the size option?

Reply:
Well, I can descope it, thought I still think we should make this change (in another PR).

Reply:
We can make a PR for the orb to adapt the call to sensei (the orb will have to be adapted anyway to offer the option to use the size parameters) and merge them together.

Reply:
I've manage to make the CLI support both positional and option arguments. We can remove the positional support when we are ready.

Reply:
Nice! I check that this evening.

Reply:
Looks like there is an issue with the default values:

```shell-session
$ sensei serve
Start dev server
Training material folder: '/formation-kubernetes-user/current working directory'. This can be changed using '--env.material=<relative path to training material folder>'.
Training slug: 'name of current working directory'. This can be changed using '--env.trainingSlug=<training slug>'.
/app/bin/sensei.js:160
  throw err;
  ^

<ref *1> Error: spawnSync /bin/sh ENOENT
    at Object.spawnSync (internal/child_process.js:1067:20)
    at spawnSync (child_process.js:602:24)
    at Object.execSync (child_process.js:647:15)
    at module.exports (/app/webpack.config.js:31:6)
    at serve (/app/bin/sensei.js:45:47)
    at cli (/app/bin/sensei.js:30:7)
    at Object.<anonymous> (/app/bin/sensei.js:177:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawnSync /bin/sh',
  path: '/bin/sh',
  spawnargs: [ '-c', 'git rev-parse --short HEAD' ],
  error: [Circular *1],
  status: null,
  signal: null,
  output: null,
  pid: 13,
  stdout: null,
  stderr: null
}
```

Proposed fix:

```patch
diff --git i/bin/sensei.js w/bin/sensei.js
index 38160d5..aa9a3c9 100755
--- i/bin/sensei.js
+++ w/bin/sensei.js
@@ -165,12 +165,12 @@ function describePositionalArguments(yargs) {
     .positional("material", {
       type: "string",
       describe: "Path to the folder containing the training material.",
-      default: "current working directory",
+      default: ".",
     })
     .positional("slug", {
       type: "string",
-      describe: "Training name used in PDF files and HTML page titles.",
-      default: "name of current working directory",
+      describe: "Training name used in PDF files and HTML page titles. (defaults to name of current working directory)",
+      default: path.basename(path.resolve()),
     });
 }
 
@@ -190,12 +190,12 @@ cli(
     .option("material", {
       type: "string",
       describe: "Path to the folder containing the training material.",
-      default: "current working directory",
+      default: ".",
     })
     .option("slug", {
       type: "string",
-      describe: "Training name used in PDF files and HTML page titles.",
-      default: "name of current working directory",
+      describe: "Training name used in PDF files and HTML page titles. (defaults to name of current working directory)",
+      default: path.basename(path.resolve()),
     })
     .option("slideWidth", {
       type: "number",
```

Reply:
From my tests, the positional arguments takes over the option, maybe we could add this info in the describe of the options?

And also add a "deprecated" info in the describe of the positional arguments.